index.js

```diff
@@ -513,15 +549,17 @@ class KeyringController extends EventEmitter {
    * encrypts that array with the provided `password`,
    * and persists that encrypted string to storage.
    *
-   * @param {string} password - The keyring controller password.
    * @returns {Promise<boolean>} Resolves to true once keyrings are persisted.
    */
-  async persistAllKeyrings(password = this.password) {
-    if (typeof password !== 'string') {
-      throw new Error('KeyringController - password is not a string');
+  async persistAllKeyrings() {
+    const { encryptionKey } = this.memStore.getState();
+
+    if (!this.password && !encryptionKey) {
```

Comment:
What's the difference between holding something in memory store vs class field? Why do we use/need both?

Reply:
Putting the `encryptionKey` in the memory store bubbles it up to the extension, which allows us to store it in session storage: https://github.com/MetaMask/metamask-extension/pull/15558/files#diff-6fbff2cfe97ac01b77296ef2122c7e0a5b3ff6a84b584b4d1a87482f35eea3d6R3941

Reply:
We don't generally have a use for _public_ class fields; the password field should be private. That's a change I was going to make in a separate PR at some point; it would be a breaking change because `@metamask/controllers` uses this field today, but they don't need to.